src/components/mgt-person-card/mgt-person-card.ts

```diff
@@ -450,6 +488,39 @@ export class MgtPersonCard extends MgtTemplatedComponent {
    * @memberof MgtPersonCard
    */
   protected async loadState() {
+    const provider = Providers.globalProvider;
+    const graph = provider.graph.forComponent(this);
+
+    // check if user is signed in
+    if (!provider || provider.state !== ProviderState.SignedIn) {
+      return;
+    }
+
+    // check if personDetail already populated
+    if (this.personDetails) {
+      // in some cases we might only have name or email, but need to find the image
+      // use @ for the image value to search for an image
+      if (this.personImage === '@' && !this.personDetails.personImage) {
+        this.loadImage();
+      }
+
+      if (this.personDetails.id) {
+        const user = this.personDetails as MicrosoftGraph.User;
+        const person = this.personDetails as MicrosoftGraph.Person;
+        const contact = this.personDetails as MicrosoftGraph.Contact;
+
+        // check if there's userPrincipalName in personDetails
+        if (user.userPrincipalName || person.userPrincipalName) {
+          // check if there's email present in personDetails
+          if (user.mail || person.scoredEmailAddresses || contact.emailAddresses) {
+            return;
+          }
+        } else {
+          this.personDetails = await getUser(graph, this.personDetails.id);
+        }
+      }
```

Comment:
We should check for `id` or `userPrincipalName`, not just `id` - both could be used interchangeably and both work the same way with the user api.

Also, you can check for the image at the same time using your new `getUserWithPhoto` function

Here is a suggested fix:

```suggestion
      const user = this.personDetails as MicrosoftGraph.User;
      const id = user.userPrincipalName || user.id;

      // if we have an id but no email, we should get data from the graph
      if (id && !getEmailFromGraphEntity(user)) {
        try {
          const person = await getUserWithPhoto(graph, id);

          this.personDetails = person;
          this.personImage = this.getImage();
        } catch {
          // nop
        }
      } else if (this.personImage === '@' && !this.personDetails.personImage) {
        // in some cases we might only have name or email, but need to find the image
        // use @ for the image value to search for an image
        this.loadImage();
      }
```